editor/doc_tools.cpp

```diff
@@ -37,12 +37,42 @@
 #include "core/io/dir_access.h"
 #include "core/io/marshalls.h"
 #include "core/object/script_language.h"
+#include "core/string/translation.h"
 #include "core/version.h"
 #include "scene/resources/theme.h"
 
 // Used for a hack preserving Mono properties on non-Mono builds.
 #include "modules/modules_enabled.gen.h" // For mono.
 
+static String _get_indent(const String &p_text) {
+	String indent;
+	bool has_text = false;
+	int line_start = 0;
+
+	for (int i = 0; i < p_text.length(); i++) {
+		const char32_t c = p_text[i];
+		if (c == '\n') {
+			line_start = i + 1;
+		} else if (c > 32) {
+			has_text = true;
+			indent = p_text.substr(line_start, i - line_start);
+			break; // Indentation of the first line that has text.
+		}
+	}
+	if (!has_text) {
+		return p_text;
+	}
+	return indent;
+}
+
+static String _translate_doc_string(const String &p_text) {
+	const String indent = _get_indent(p_text);
+	const String message = p_text.dedent().strip_edges();
+	const String translated = TranslationServer::get_singleton()->doc_translate(message, "");
+	// No need to restore stripped edges because they'll be stripped again later.
+	return translated.indent(indent);
+}
```

Comment:
In general I feel like we could be doing a better job at handling the XML tab indentation of these strings. The only part which we need to preserve is the manual 4-space indentation in code blocks, but everything else should be something that we could simply ignore, and add manually in `_write_string` (as already done at least one the first line).

Currently we dedent, strip edges, indent again, strip edges again (so it removes indentation on the first line again?), and then escape. A lot of steps which could be done better IMO.

But that's not blocking for this PR.